### PR TITLE
chore: Reorder variants of `KnownHyperlaneDomain`

### LINKED ISSUE #3961: improve alphabetical ordering of `KnownHyperlaneDomain` fields
we still want to keep the separation between testnets and mainnets, but would be good to have the mainnets ordered alphabetically.

raised by @paulbalaji in https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3960#discussion_r1638705840

### PR DESCRIPTION
**Pull request by c410-f3r**
## Description

Fix #3961

### CODE REVIEW

**Comment by paulbalaji** [comment]
hey @c410-f3r would you be able to update the PR after pulling the latest changes from main please?

**Comment by c410-f3r** [comment]
Done

**Comment by paulbalaji** [comment]
> Done

thanks! would be great if you could also sort where the enum is used below:

- `pub const fn domain_type(self) -> HyperlaneDomainType`
- `pub const fn domain_protocol(self) -> HyperlaneDomainProtocol`
- `pub const fn domain_technical_stack(self) -> HyperlaneDomainTechnicalStack`

**Comment by c410-f3r** [comment]
Done

File: rust/hyperlane-core/src/chain.rs [hunk 1]
```diff
@@ -258,16 +228,17 @@ impl KnownHyperlaneDomain {
...
+                BinanceSmartChainTestnet, Blast, Bob, Celo, Chiado, Ethereum, Fraxtal, Fuji,
+                Gnosis, Holesky, InEvm, Linea, MantaPacific, Mantle, Mode, MoonbaseAlpha, Moonbeam,
+                Optimism, PlumeTestnet, Polygon, Redstone, ScrollSepolia, Sei, Sepolia, Taiko,
+                Test1, Test2, Test3, Viction, Zetachain,
             ],
             HyperlaneDomainProtocol::Fuel: [FuelTest1],
             HyperlaneDomainProtocol::Sealevel: [SealevelTest1, SealevelTest2],
-            HyperlaneDomainProtocol::Cosmos: [CosmosTest99990, CosmosTest99991, Neutron, Osmosis, Injective],
+            HyperlaneDomainProtocol::Cosmos: [
+                CosmosTest99990, CosmosTest99991, Injective, Neutron, Osmosis
```
Comment (paulbalaji) [inline rust/hyperlane-core/src/chain.rs:240]: would be great to separate by mainnet/testnet here (just a comment dividing them is fine, similar to how the top-level enum is ordered)

Comment (paulbalaji) [reply rust/hyperlane-core/src/chain.rs:240]: ditto lines 251-257

**Comment by paulbalaji** [comment]
almost there! please run `cargo fmt` from the `rust` dir

**Comment by c410-f3r** [comment]
Looks like a flaky test -> https://github.com/hyperlane-xyz/hyperlane-monorepo/actions/runs/9785409597/job/27048090453

**Comment by paulbalaji** [comment]
> Looks like a flaky test -> [hyperlane-xyz/hyperlane-monorepo/actions/runs/9785409597/job/27048090453](https://github.com/hyperlane-xyz/hyperlane-monorepo/actions/runs/9785409597/job/27048090453)

yup that one can flake sometimes, dw about it :)

**Review by paulbalaji — APPROVED** [review]

**Comment by avious00** [comment]
hey @c410-f3r thanks for your contribution here! if there's other bounties you're interested in let us know, we have compensation between $250-$1000 typically

**Comment by c410-f3r** [comment]
Well, I will gladly continue with my contributions after the merge of https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4104.

**Pull request merged**